### PR TITLE
Add exclude pattern to the quasar agent

### PR DESCRIPTION
**Pull request by exFalso**
The pattern was generated with the quasar-hook tool and should reduce node startup time considerably (16-17 seconds -> 9-10 seconds on my laptop)

### CODE REVIEW

File: node/capsule/build.gradle [hunk 1]
```diff
@@ -36,10 +36,8 @@ task buildCordaJAR(type: FatCapsule, dependsOn: project(':node').compileJava) {
...
-        // def quasarExcludeExpression = "x(rx**;io**;kotlin**;jdk**;reflectasm**;groovyjarjarasm**;groovy**;joptsimple**;groovyjarjarantlr**;javassist**;com.fasterxml**;com.typesafe**;com.google**;com.zaxxer**;com.jcabi**;com.codahale**;com.esotericsoftware**;de.javakaffee**;org.objectweb**;org.slf4j**;org.w3c**;org.codehaus**;org.h2**;org.crsh**;org.fusesource**;org.hibernate**;org.dom4j**;org.bouncycastle**;org.apache**;org.objenesis**;org.jboss**;org.xml**;org.jcp**;org.jetbrains**;org.yaml**;co.paralleluniverse**;net.i2p**)"
-        // javaAgents = ["quasar-core-${quasar_version}-jdk8.jar=${quasarExcludeExpression}"]
-        javaAgents = ["quasar-core-${quasar_version}-jdk8.jar"]
+        def quasarExcludeExpression = "x(rx**;io**;antlr**;kotlin**;jdk**;reflectasm**;groovyjarjarasm**;groovy**;joptsimple**;groovyjarjarantlr**;com.fasterxml**;com.typesafe**;com.google**;com.zaxxer**;com.jcabi**;com.codahale**;com.esotericsoftware**;de.javakaffee**;org.objectweb**;org.slf4j**;org.w3c**;org.codehaus**;org.crsh**;org.fusesource**;org.h2**;org.hibernate**;org.dom4j**;org.bouncycastle**;org.apache**;org.objenesis**;org.jboss**;org.jcp**;org.xml**;co.paralleluniverse**;net.i2p**)"
```
Comment (mikehearn) [inline node/capsule/build.gradle]: Could you add a comment explaining what this is and how it was generated/how to update it?

**Comment by mikehearn** [comment]
Can we also add an exclusion for org.jolokia to avoid an issue @Quiark has reported?

File: node/capsule/build.gradle (hunk 1 above)
Comment (mikehearn) [reply node/capsule/build.gradle]: Also: co.paralleluniverse? Really?!

**Comment by Quiark** [comment]
Besides some Jolokia classes, I also saw it complain about `kotlin/jvm/internal/Intrinsics`,`javax/management/MBeanServerConnection`

**Comment by exFalso** [comment]
I generated the pattern using experimental/quasar-hook, starting up a node and then immediately quitting. The tool creates the pattern using the set of loaded classes - this means we should exercise the node as much as we can to cover most (non-)use cases, naively starting and quitting is not enough perhaps.

I'll try to run the tool on the full unit test suite, that should cover those cases as well I reckon. I'll try to add a run config as well for future use

File: node/capsule/build.gradle (hunk 1 above)
Comment (exFalso) [reply node/capsule/build.gradle]: Yeah, well, it *scans* everything, but only instruments a small subset

**Comment by exFalso** [comment]
I added a run config that inspects classes loaded by `:node` tests, this added more patterns.
Also documented how to generate a pattern, it's not a simple one liner, but I didn't want to spend too much time on cleaning it up, especially as we may change it altogether.

Re jolokia: as it's an external library that corda doesn't use by default I don't think it's a good idea to add it manually to the exclude pattern. Instead we could file an issue for this, and dedicate some time later on. As discussed on slack the proper solution would be to restrict the instrumentation based on classloaders rather than these patterns, alternatively we could expose a flag that extends the pattern. I didn't want to spend more time on this as this is mostly a convenience feature

**Pull request merged**